### ISSUE TITLE
[Bug]: Errors in Gathering Space Group Info For Materials Project 

### ISSUE THREAD

**Issue by melo-gonzo** [IntelLabs/matsciml#50]
### Expected behavior

Using the `get_space_group_info()` should not result in errors. 

### Actual behavior

The `get_space_group_info()` method called [here](https://github.com/melo-gonzo/matsciml/blob/38e1d647afcffa619b42fbd4267c21e57d7aa6cb/matsciml/datasets/materials_project/dataset.py#L129) gives various errors:

```python
spglib: ssm_get_exact_positions failed (attempt=1). (line 115, /project/src/site_symmetry.c).
spglib: ssm_get_exact_positions failed (attempt=2). (line 115, /project/src/site_symmetry.c).
spglib: ssm_get_exact_positions failed (attempt=3). (line 115, /project/src/site_symmetry.c).
###
spglib: Attempt 0 tolerance = 1.000000e-02 failed(line 800, /project/src/spacegroup.c).
spglib: No point group was found (line 405, /project/src/pointgroup.c).
spglib: Attempt 1 tolerance = 9.500000e-03 failed(line 800, /project/src/spacegroup.c).
###
spglib: Too many lattice symmetries was found.
        Reduce angle tolerance to 4.286875
        (line 990, /project/src/symmetry.c).
```

### Steps to reproduce the problem

Install the repo, and run this scrip with the full materials project dataset: 

`python examples/datasets/materials_project/single_task_symmetry.py`

### Specifications

Using v1.0.0, error occurs on both cpu/gpu. 

Pip Freeze:
aiohttp==3.8.5
aiosignal==1.3.1
ase==3.22.1
async-timeout==4.0.3
attrs==23.1.0
certifi==2023.7.22
charset-normalizer==3.2.0
cloudpickle==2.2.1
contextlib2==21.6.0
contourpy==1.1.0
cycler==0.11.0
dgl==0.9.1
dgllife==0.3.2
docstring-parser==0.15
emmet-core==0.68.0
exceptiongroup==1.1.3
filelock==3.12.4
fonttools==4.42.1
frozenlist==1.4.0
fsspec==2023.9.0
future==0.18.3
geometric-algebra-attention==0.4.0
hyperopt==0.2.7
idna==3.4
importlib-resources==6.0.1
iniconfig==2.0.0
Jinja2==3.1.2
joblib==1.3.2
jsonargparse==4.24.1
kiwisolver==1.4.5
latexcodec==2.0.1
lightning-utilities==0.9.0
llvmlite==0.40.1
lmdb==1.3.0
MarkupSafe==2.1.3
matplotlib==3.7.3
-e git+https://github.com/melo-gonzo/matsciml.git@db46be635795d4ec556dc00222058a61909d0cf7#egg=matsciml
monty==2023.9.5
mp-api==0.34.3
mpmath==1.3.0
msgpack==1.0.5
multidict==6.0.4
munch==2.5.0
networkx==3.1
numba==0.57.1
numpy==1.24.4
packaging==23.1
palettable==3.3.3
pandas==2.1.0
Pillow==10.0.0
plotly==5.16.1
pluggy==1.3.0
protobuf==4.24.3
psutil==5.9.5
py4j==0.10.9.7
pybtex==0.24.0
pydantic==1.10.12
pymatgen==2023.9.10
pyparsing==3.1.1
pytest==7.4.2
python-dateutil==2.8.2
pytorch-lightning==1.8.6
pytz==2023.3.post1
PyYAML==6.0.1
rdkit==2023.3.1
requests==2.31.0
ruamel.yaml==0.17.32
ruamel.yaml.clib==0.2.7
schema==0.7.5
scikit-learn==1.3.0
scipy==1.11.2
six==1.16.0
spglib==2.1.0
sympy==1.12
tabulate==0.9.0
tenacity==8.2.3
tensorboardX==2.6.2.2
threadpoolctl==3.2.0
tomli==2.0.1
torch==2.0.1
torchmetrics==1.1.2
tqdm==4.66.1
typeshed-client==2.3.0
typing_extensions==4.7.1
tzdata==2023.3
uncertainties==3.1.7
urllib3==2.0.4
yarl==1.9.2
zipp==3.16.2
